src/main/java/io/quarkus/registry/app/util/Version.java

```diff
@@ -22,15 +22,23 @@ public static void validateVersion(String version) {
 
     /**
      * Order versions based on the qualifier. Final > CR1
+     *
+     * 2.5.0.Final > 2.6.0.CR1
+     * 2.6.0.SP1 > 2.5.0.Final
      */
-    public static final Comparator<String> QUALIFIER_REVERSED_COMPARATOR = ((o1, o2) -> {
-        String o1Qualifier = new DefaultArtifactVersion(o1).getQualifier();
-        String o2Qualifier = new DefaultArtifactVersion(o2).getQualifier();
-        if (o1Qualifier == null || o2Qualifier == null) {
-            return 0;
-        } else {
-            return o2Qualifier.compareTo(o1Qualifier);
+    public static final Comparator<String> RELEASE_IMPORTANCE_COMPARATOR = ((o1, o2) -> {
+        DefaultArtifactVersion v1 = new DefaultArtifactVersion(o1);
+        DefaultArtifactVersion v2 = new DefaultArtifactVersion(o2);
+        String o1Qualifier = v1.getQualifier();
+        String o2Qualifier = v2.getQualifier();
+        if (o1Qualifier != null && o2Qualifier != null) {
+            if (o1Qualifier.startsWith("CR")) {
+                return 1;
```

Comment:
This doesn't look right. We know `o2Qualifier` is not null but we haven't compared it yet.

Reply:
Good catch, I updated it and improved the test